lib/yt/models/snippet.rb

```diff
@@ -51,6 +51,11 @@ def can_reply?
         @can_reply ||= data.fetch 'canReply', false
       end
 
+      def top_level_comment
+        top_level_comment = ActiveSupport::HashWithIndifferentAccess.new data['topLevelComment']
+        @top_level_comment ||= Yt::Comment.new top_level_comment
```

Comment:
How about `@top_level_comment ||= Yt::Comment.new data['topLevelComment'].symbolize_keys` ?

Would that work?